tasks/homework/homework_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestMoveAnimal(t *testing.T) {
+	// Arrange
+	z := Zoo{
+		Areas: buildAreas(),
+	}
+
+	ungulatesAnimals := z.Areas["ungulates"].Sectors["animals"]
+	newSector := Sector{
+		Subtype: "animals",
+		Animals: []Animal{},
+	}
+
+	// Act
+	ungulatesAnimals.MoveAnimal(2, &newSector)
+
+	// Assert
+	if len(ungulatesAnimals.Animals) != 2 {
+		t.Errorf("expected 2 animals, got %d", len(ungulatesAnimals.Animals))
+	}
+
+	if len(newSector.Animals) != 1 {
+		t.Errorf("expected 1 animal in new sector, got %d", len(newSector.Animals))
+	}
+
+	if newSector.Animals[0].Name != "Horse" {
+		t.Errorf("expected 'Horse' in new sector, got %s", newSector.Animals[0].Name)
+	}
+}
+
+func TestFindAnimalByNameSuccessful(t *testing.T) {
+	// Arrange
+	z := Zoo{
+		Areas: buildAreas(),
+	}
+	// Act
+	_, err := FindAnimalByName(z.Areas, "Eagle")
+	// Assert
+	if nil != err {
```

Comment:
Ось це єдино правильна конструкція в коді та тестах:
```suggestion
	if err != nil {
```